scripts/localVsDissecLearning.js

```diff
@@ -0,0 +1,238 @@
+const fs = require('fs')
+const { v4: uuid } = require('uuid')
+const {
+  default: CozyClient,
+  createClientInteractive,
+  Q
+} = require('cozy-client')
+
+const { BANK_DOCTYPE } = require('../src/doctypes/bank')
+const { JOBS_DOCTYPE } = require('../src/doctypes/jobs')
+const { NODES_DOCTYPE } = require('../src/doctypes/nodes')
+const { PERFORMANCES_DOCTYPE } = require('../src/doctypes/performances')
+const dissecConfig = require('../dissec.config.json')
+
+const aggregationNodes = require('../assets/webhooks.json')
+const { exit } = require('process')
+
+/**
+ * This script measures performances of DISSEC vs local learning.
+ *
+ * It supposes that the querier instance has at least 2 categories of bank operations.
+ */
+
+const runExperiment = async () => {
+  const uri = process.argv[2]
+  if (!uri)
+    throw new Error('Expected the URI of the executing instance as parameter')
+
+  // Helper
+  const getCategory = doc => {
+    return doc.manualCategoryId || doc.localCategoryId || doc.cozyCategoryId
+  }
+
+  // Connect to the instance
+  const client = await createClientInteractive({
+    scope: [BANK_DOCTYPE, JOBS_DOCTYPE],
+    uri: process.argv[2],
+    schema: {
+      operations: {
+        doctype: BANK_DOCTYPE,
+        attributes: {},
+        relationships: {}
+      }
+    },
+    oauth: {
+      softwareID: 'io.cozy.client.cli'
+    }
+  })
+
+  // Download all bank operations
+  const allOperations = await client.queryAll(Q(BANK_DOCTYPE))
+
+  // Filter and update data
+  const allCategories = allOperations.map(e => getCategory(e))
+  const uniqueCategories = []
+  allCategories.forEach(
+    e => !uniqueCategories.includes(e) && uniqueCategories.push(e)
+  )
+
+  const sortedOperations = allOperations.sort(
+    (a, b) =>
+      new Date(a.cozyMetadata.createdAt).valueOf() -
+      new Date(b.cozyMetadata.createdAt).valueOf()
+  )
+  const cutoffDate = new Date(
+    sortedOperations[
+      Math.round(sortedOperations.length / 2)
+    ].cozyMetadata.createdAt
+  )
+  const validationSet = sortedOperations
+    .filter(
+      e =>
+        cutoffDate.valueOf() - new Date(e.cozyMetadata.createdAt).valueOf() > 0
+    )
+    .sort((a, b) => a.id - b.id)
```

Comment:
Why not just slicing the operations in the middle? 
```
const sortedOperations = await client.queryAll(Q(BANK_DOCTYPE).sortBy([{"date": "asc"}])`
const validationSet = sortedOperations.splice(0, Math.round(sortedOperations.length / 2))
```

Reply:
Also, it seems a bit counter-intuitive to me to have the most ancient operations in the validation set, and the most recent in the categorization set. The reverse would make more sense, don't you think?